src/ray/protobuf/experimental/instance_manager.proto

```diff
@@ -0,0 +1,143 @@
+// period (by default 30 minutes).
+message Instance {
+  enum InstanceStatus {
+    // The unspecified state - most likey it is queued.
+    INSTANCE_STATUS_UNSPECIFIED = 0;
+    // Instance is starting. The first state update received from the
+    // instance.
+    STARTING = 1;
+    // The instance is running - one of two states of a healthy instance.
+    RUNNING = 2;
+    // The instance is idle - one of two states of a healthy instance.
+    IDLE = 3;
+    // The instance is stopping - usually follows from the RUNNING, IDLE,
+    // PREEMPT_REQUEST or DRAIN_REQUEST state.
+    STOPPING = 4;
+    // The instance is stopped - follows from the STOPPING state.
+    STOPPED = 5;
+    // The instance is in a bad state - but it is still able to send updates.
+    FAILING = 6;
+    // The subscribe service moves instances to this state if they
+    // have been idle for too long. This allows the cluster manager to
+    // make a final decision on whether or not to commence a drain
+    // sequence for this instance.
+    DRAIN_CONFIRMATION_PENDING = 7;
+    // The instance should be drained, Ray should start draining process
+    // but could reject if failed to drain.
+    DRAIN_REQUEST = 8;
+    // The instance will be reempted by the instance manager, regardless
+    // of whether it is drainable or not.
+    PREEMPT_REQUEST = 9;
+    // An optional state that can be used to indicate that the instance
+    // is allocated from cloud provider, but ray hasn't been installed yet.
+    INSTANCE_ALLOCATED = 10;
+    // An optional state that can be used to indicate that the instance
+    // is currently installing Ray.
+    INSTALLING_RAY = 11;
+    // An optional state that can be used to indicate that the instance
+    // failed to allocate from cloud provider.
+    ALLOCATION_FAILED = 12;
+    // Node is deleted.
+    GARAGE_COLLECTED = 13;
+  }
+  // an unique id for the instance that's generated by the
+  // instance manager. This may be optional if
+  // the instance hasn't be started yet.
+  string instance_id = 11;
+  // the status of the instance.
+  InstanceStatus status = 12;
+  // the node id of the instance.
+  string node_type = 13;
+  // The corresponding total resources on the node.
+  map<string, double> total_resources = 14;
+  // timestamp of the last state changed.
+  int64 timestamp_since_last_state_change = 15;
+  // the external id of the instance that's generated by
+  // the cloud provider like AWS, GCP, etc.
+  // Note this id can be reused by different instances.
+  string cloud_instance_id = 16;
+  // internal ip address of the instance.
+  string internal_ip = 17;
+  // external ip address of the instance.
+  string external_ip = 18;
+  // the monotonically increasing version number of the instance.
+  int64 version = 19;
+}
+
+message UpdateInstanceManagerStateRequest {
+  int64 expected_version = 1;
+  repeated InstanceType new_nodes_to_start = 2; 
+  repeated bytes nodes_to_terminate = 3;
```

Comment:
Is it supposed to be the type_name? or instance_id? If it is instance_id, the type seems to be incorrect (instance_id is string).

Also is it possible to use the same name as in other protobuf to make this clear? (e.g., if this is instance_id, can you use instance_ids_to_terminate?)